### PR TITLE
refactor: use `simulate` instead of `get` on `BaseInvocationScope` class

### PR DESCRIPTION
**Pull request by Torres-ssf**
Follow-up from https://github.com/FuelLabs/fuels-ts/issues/1093

### CODE REVIEW

**Review by danielbate — DISMISSED** [review]
:shipit:

File: apps/docs/src/guide/contracts/simulate-transactions.md [hunk 1]
```diff
@@ -0,0 +1,46 @@
+# Simulate Transactions
+
+<!-- This section should explain simulate transactions -->
+<!-- simulate:example:start -->
+
+Simulating a transaction is a powerful feature that allows you to validate whether a given transaction meets its requirements without actually modifying the blockchain. It's a safe way to ensure that a transaction will behave as expected without spending any real resources.
```
Comment (arboleya) [inline apps/docs/src/guide/contracts/simulate-transactions.md:6]:
```suggestion
Simulating a transaction is a powerful feature that allows you to validate whether a given transaction meets its requirements without modifying the blockchain. It's a safe way to ensure that a transaction will behave as expected without spending any real resources.
```

File: apps/docs/src/guide/contracts/simulate-transactions.md [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+Simulating a transaction allows you to catch potential errors or issues before submitting the real transaction, making it a valuable tool for development and user interaction.
+
+<!-- simulate:example:end -->
+
+## Read-only Calls Using `simulate()`
+
+<!-- This section should explain read-only calls -->
+<!-- read:example:start -->
+
+When interacting with a contract, you might want to execute a method that does not change the state of the blockchain. For example, reading a value from storage without modifying it.
```
Comment (arboleya) [inline apps/docs/src/guide/contracts/simulate-transactions.md:21]:
```suggestion
When interacting with a contract, you might want to execute a method that does not change the state of the blockchain—for example, reading a value from storage without modifying it.
```

File: apps/docs/src/guide/contracts/simulate-transactions.md [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+In such cases, you can use the `simulate()` method:
+
+<<< @/../../docs-snippets/src/guide/contracts/simulate-transactions.test.ts#simulate-transactions-2{ts:line-numbers}
+
+Using `simulate()` for read-only calls or to test transactions provides a way to safely and efficiently access information from the blockchain or validate how a transaction would behave. It ensures that no changes will be made to the blockchain and no resources will be spent, making it a valuable tool for both data retrieval and development.
```
Comment (arboleya) [inline apps/docs/src/guide/contracts/simulate-transactions.md:27]:
```suggestion
Using `simulate()` for read-only calls or to test transactions provides a way to safely and efficiently access information from the blockchain or validate how a transaction would behave. It ensures that no changes will be made to the blockchain and no resources will be spent, making it a valuable tool for data retrieval and development.
```

File: apps/docs/src/guide/contracts/simulate-transactions.md [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+
+<!-- read:example:end -->
+
+## When to use `simulate()` vs `call()`
+
+<!-- This section should explain when to use the simulate vs call methods -->
+<!-- simulate_call:example:start -->
+
+`simulate()`: Suitable for both read-only calls and testing transactions that could modify the blockchain's state, this method allows you to safely validate how functions will behave without actually making changes. It's a powerful tool for validation, development, and data retrieval, providing confidence in your transactions and interactions with the blockchain.
```
Comment (arboleya) [inline apps/docs/src/guide/contracts/simulate-transactions.md:36]:
```suggestion
`simulate()`: Suitable for both read-only calls and testing transactions that could modify the blockchain's state, this method allows you to safely validate how functions will behave without making changes. It's a robust validation, development, and data retrieval utility method, providing confidence in your transactions and interactions with the blockchain.
```

File: apps/docs/src/guide/contracts/simulate-transactions.md [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+
+`call()`: Use this method when calling a function that modifies the blockchain's state, such as transferring funds or updating values. Keep in mind that if you use `call()` on a contract method that changes the blockchain state, it WILL spend your resources.
```
Comment (arboleya) [inline apps/docs/src/guide/contracts/simulate-transactions.md:38]:
```suggestion
`call()`: Use this method when calling a function that modifies the blockchain's state, such as transferring funds or updating values. Remember that if you use `call()` on a contract method that changes the blockchain state, it _**will**_ spend your resources.
```

File: apps/docs/src/guide/contracts/simulate-transactions.md [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+
+<!-- simulate_call:example:end -->
+
+---
+
+It is worth noting that while `call()` can be used to execute read-only contract methods without spending resources, it's recommended to use `simulate()` for such calls. Using `simulate()` provides a guarantee that even if you accidentally execute a contract method that would normally change the blockchain, no resources will be spent. This additional safeguard makes `simulate()` a preferable choice for read-only interactions.
```
Comment (arboleya) [inline apps/docs/src/guide/contracts/simulate-transactions.md:44]:
```suggestion
It is worth noting that while `call()` can be used to execute read-only contract methods without spending resources, it's recommended to use `simulate()` for such calls. Using `simulate()` guarantees that no resources will be spent even if you accidentally execute a contract method that would typically change the blockchain. This additional safeguard makes `simulate()` preferable for read-only interactions.
```

File: apps/docs/src/guide/contracts/simulate-transactions.md [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+
+> **Note:** Even though a `simulate()` call won't modify the blockchain's state and does not spent resources, the transaction must still meet its requirements to be considered valid.
```
Comment (arboleya) [inline apps/docs/src/guide/contracts/simulate-transactions.md:46]:
```suggestion
> **Note:** Although a `simulate()` call won't modify the blockchain's state and does not spend resources, the transaction must still meet its requirements to be considered valid.
```

**Review by arboleya — CHANGES_REQUESTED** [review]
Small nits. Feel free to accept them selectively.

**Review by arboleya — APPROVED** [review]

**Review by Dhaiwat10 — APPROVED** [review]

**Pull request merged**